### PR TITLE
added support for requesting new access_token with refresh_token && new calls to get people, individual user, and user hours

### PR DESCRIPTION
**Pull request by memmott**
@dlo this repo has not been updated in awhile. Do you plan to continue updating?

### CODE REVIEW

**Comment by dlo** [comment]
Unfortunately, I don't have the PyPi deployment key so can't push live.

@gxela thoughts? I'm happy to take over pushing.

PS Changes look great!

File: harvest/harvest.py [hunk 1]
```diff
@@ -301,8 +311,8 @@ def _request(self, method='GET', path='/', data=None):
                 try:
                     return resp.json()
                 except:
-                    return resp
-            return resp
+                    return resp.json()
```
Comment (maartenkling) [inline harvest/harvest.py]: why try expect if you do the same thing twice?

Comment (memmott) [inline harvest/harvest.py]: I was experimenting with output types and forgot to change this back. Fixed

**Comment by memmott** [comment]
@maartenkling happy to merge with local changes and/or add anything... is there anything else you need for the PR to be merged? This authentication is working in production for me.

File: harvest/harvest.py [hunk 2]
```diff
@@ -301,7 +311,7 @@ def _request(self, method='GET', path='/', data=None):
                 try:
                     return resp.json()
                 except:
-                    return resp
+                    return resp.json()
```
Comment (maartenkling) [inline harvest/harvest.py]: now the try and except are the same

**Comment by maartenkling** [comment]
@memmott please see my comment

**Comment by memmott** [comment]
:+1:

File: harvest/harvest.py [hunk 3]
```diff
@@ -20,17 +12,21 @@ class HarvestError(Exception):
...
+    def __init__(self, uri, email=None, password=None, refresh_token=None, client_id=None, token=None, put_auth_in_header=True):
         self.__uri = uri.rstrip('/')
         parsed = urlparse(uri)
         if not (parsed.scheme and parsed.netloc):
             raise HarvestError('Invalid harvest uri "{0}".'.format(uri))
-
-        self.__headers = {
-            'Content-Type'  : 'application/json',
-            'Accept'        : 'application/json',
-            'User-Agent'    : 'Mozilla/5.0',  # 'TimeTracker for Linux' -- ++ << >>
```
Comment (maartenkling) [inline harvest/harvest.py]: this line is missing now

Comment (memmott) [inline harvest/harvest.py]: It is not missing. It checks whether it contains refresh_token, else it uses the normal self._headers. I did remove the 'User-Agent': ... but added back for consistency.

File: README.md [hunk 4]
```diff
@@ -2,30 +2,24 @@
...
+Token must look like this:
+
+token = { 
+	'token_type': 'bearer', 
+	'access_token': 'your access token', 
+	'refresh_token': 'your refresh token',
+	'expires_in': 64799,
+}
+
+For information on how to get intial tokens see: https://github.com/harvesthq/api/blob/master/Authentication/OAuth%202.0.md
```
Comment (trikosuave) [inline README.md]: initial*

**Comment by dlo** [comment]
Closes #12

**Comment by dlo** [comment]
@memmott Thanks for your contributions!

File: README.md (hunk 4 above)
Comment (dlo) [inline README.md]: Fixed in 9c070bb

**Pull request merged**